string.c

```diff
@@ -1556,21 +1556,82 @@ rb_str_resurrect(VALUE str)
 
 /*
  *  call-seq:
- *     String.new(str="")                   -> new_str
- *     String.new(str="", encoding: enc)    -> new_str
- *     String.new(str="", capacity: size)   -> new_str
- *
- *  Returns a new string object containing a copy of <i>str</i>.
- *
- *  The optional <i>encoding</i> keyword argument specifies the encoding
- *  of the new string.
- *  If not specified, the encoding of <i>str</i> is used
- *  (or ASCII-8BIT, if <i>str</i> is not specified).
- *
- *  The optional <i>capacity</i> keyword argument specifies the size
- *  of the internal buffer.
- *  This may improve performance, when the string will be concatenated many
- *  times (causing many realloc calls).
+ *     String.new(str='')                   -> new_str
+ *     String.new(str='', encoding: enc)    -> new_str
+ *     String.new(str='', capacity: size)   -> new_str
+ *
+ *  If argument +str+ is given, it must be a
+ *  {String-convertible object}[doc/implicit_conversion_rdoc.html#label-String-Convertible+Objects]
+ *  (implements +to_str).
+ *
+ *  Argument +encoding+, if given, must be:
+ *  - A {String-convertible object}[doc/implicit_conversion_rdoc.html#label-String-Convertible+Objects]
+ *    (implements +to_str+).
+ *  - The name of an encoding that is compatible with +str+.
```

Comment:
I think "encoding" should be capitalized to link to the Encoding documentation either here or below where the `encoding:` argument and String.new default encodings are discussed

Reply:
Done below.